src/Orleans/Runtime/GrainReference.cs

```diff
@@ -407,6 +407,8 @@ private static void ResponseCallback(Message message, TaskCompletionSource<objec
             {
                 context.TrySetException(response.Exception);
             }
+
+            message.Dispose();
```

Comment:
I think we'd leak if message.RejectionMessage == RejectionTypes.DuplicateRequest